components/Toaster.tsx

```diff
@@ -0,0 +1,147 @@
+import { useSyncExternalStore } from 'react'
+import { Animated, View } from 'react-native'
+import { AnimatedFlex } from './common/ui/Flex'
+import { Text } from './common/ui/Text'
+
+type Toast = {
+  id: number
+  text: string
+  duration?: number
+  animatedValue?: Animated.Value
+}
+
+type Action =
+  | {
+      type: 'ADD'
+      toast: Toast
+    }
+  | {
+      type: 'REMOVE'
+      id: number
+    }
+
+const TOAST_LIMIT_POLICY = 5
+
+let toastMemory: Toast[] = []
+let listeners: ((toasts: Toast[]) => void)[] = []
+
+const reducer = (state: Toast[], action: Action): Toast[] => {
+  switch (action.type) {
+    case 'ADD':
+      const newState = [...state, action.toast]
+      return newState.slice(0, TOAST_LIMIT_POLICY)
```

Comment:
[Suggestion]

5개 이상에서는 새로운 토스트가 잘릴 수 있을 것 같아요.
아니면 제가 잘못 이해한 건지 모르겠네요.


```suggestion
    case 'ADD':
      const newState = [...state, action.toast]
      if (newState.length > TOAST_LIMIT_POLICY) { 
        return newState.slice(-TOAST_LIMIT_POLICY)
      }
      return newState
```

Reply:
5개로 slice 한건 의도한 부분이긴 한데, 저희 서비스에서는 토스트가 5개 이상 나올 확률은 없긴 한 것 같아요
당장 큰 문제는 없을 것 같습니다!